### PR TITLE
SC-2047 Remove Interservicecommunication dependency

### PR DESCRIPTION
**Pull request by reshmi-nair**

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/email/impl/EmailServiceClientImpl.java [hunk 1]
```diff
@@ -23,18 +26,27 @@ public Response sendMail(ActorRef actorRef, Map<String, Object> requestMap) {
...
-      throw (ProjectCommonException) obj;
-    } else {
-      throw new ProjectCommonException(
-          ResponseCode.SERVER_ERROR.getErrorCode(),
-          ResponseCode.SERVER_ERROR.getErrorMessage(),
-          ResponseCode.SERVER_ERROR.getResponseCode());
+      if (obj instanceof Response) {
+        response = (Response) obj;
+      } else if (obj instanceof ProjectCommonException) {
+        throw (ProjectCommonException) obj;
```
Comment (amit-tarento) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/email/impl/EmailServiceClientImpl.java:38]: No need to else if and else condition if we are catching all exception and throwing differnet exception like unableToCommunicateWithActor exception.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/location/impl/LocationClientImpl.java [hunk 2]
```diff
@@ -152,6 +151,24 @@ public void updateLocation(
     return new ArrayList<>();
   }
 
+  private Object actorCall(ActorRef actorRef, Request request, RequestContext context){
+    Object obj = null;
+    try {
+      Timeout t = new Timeout(Duration.create(10, TimeUnit.SECONDS));
+      Future<Object> future = Patterns.ask(actorRef, request, t);
+      obj = Await.result(future, t.duration());
+      checkLocationResponseForException(obj);
```
Comment (amit-tarento) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/location/impl/LocationClientImpl.java:160]: This check not required.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java [hunk 3]
```diff
@@ -67,20 +69,27 @@ private String upsertOrg(
...
-          ResponseCode.SERVER_ERROR.getResponseCode());
+    try {
+      Timeout t = new Timeout(Duration.create(10, TimeUnit.SECONDS));
+      Future<Object> future = Patterns.ask(actorRef, request, t);
+      Object obj = Await.result(future, t.duration());
+
+      if (obj instanceof Response) {
+        Response response = (Response) obj;
+        orgId = (String) response.get(JsonKey.ORGANISATION_ID);
+      } else if (obj instanceof ProjectCommonException) {
```
Comment (amit-tarento) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java:80]: else if and else condition not required

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/email/impl/EmailServiceClientImpl.java (hunk 1 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/email/impl/EmailServiceClientImpl.java]: Here we are checking the response obj type and throwing the exception. Added Projectcommon exception throw clause in try catch

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/location/impl/LocationClientImpl.java (hunk 2 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/location/impl/LocationClientImpl.java]: Here we are checking the response obj type and throwing the exception, so thinking to keep it for now.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java (hunk 3 above)
Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java]: Here we are checking the response obj type and throwing the exception, so thinking to keep it for now.

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java [hunk 4]
```diff
@@ -96,26 +105,33 @@ public Organisation getOrgById(ActorRef actorRef, String orgId, RequestContext c
...
+      if (obj instanceof Response) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Response response = (Response) obj;
+
+        // Convert contact details (received from ES) format from map to
+        // JSON string (as in Cassandra contact details are stored as text)
+        Map<String, Object> map = (Map) response.get(JsonKey.RESPONSE);
+        map.put(JsonKey.CONTACT_DETAILS, String.valueOf(map.get(JsonKey.CONTACT_DETAILS)));
+        organisation = objectMapper.convertValue(map, Organisation.class);
+      } else if (obj instanceof ProjectCommonException) {
```
Comment (amit-tarento) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/actor-util/src/main/java/org/sunbird/actorutil/org/impl/OrganisationClientImpl.java]: its better to just check for obj instanceof Response and move all remaining check to catch block 
like
catch (Exception e) {
 if (obj instanceof ProjectCommonException) {
      throw (ProjectCommonException) obj;
    } else if (obj instanceof Exception) {
      throw new ProjectCommonException(
          ResponseCode.SERVER_ERROR.getErrorCode(),
          ResponseCode.SERVER_ERROR.getErrorMessage(),
          ResponseCode.SERVER_ERROR.getResponseCode());
    }
}

**Review by amit-tarento — APPROVED** [review]

**Pull request merged**